vm/systemSmartContracts/auction.go

```diff
@@ -79,6 +79,12 @@ func NewStakingAuctionSmartContract(
 	if check.IfNil(args.Marshalizer) {
 		return nil, vm.ErrNilMarshalizer
 	}
+	if check.IfNil(args.SigVerifier) {
+		return nil, vm.ErrNilMessageSignVerifier
+	}
+	if args.GenesisTotalSupply == nil || args.GenesisTotalSupply.Cmp(zero) <= 0 {
+		return nil, fmt.Errorf("%w, value is %v", vm.ErrInvalidGenesisTotalSupply, args.GenesisTotalSupply)
+	}
```

Comment:
L210 is incompatible with the mainnet running version:
mainnet:
```GO
s.eei.AddReturnMessage("could not get all blsKeys from registration data: error " + err.Error())
err = errors.New("public key missmatch")
```

here:
```GO
s.eei.AddReturnMessage("could not get all blsKeys from registration data: " + vm.ErrBLSPublicKeyMissmatch.Error())
var ErrBLSPublicKeyMissmatch = errors.New("public key missmatch")
```

Reply:
pushed